File: SellGUI-DynaShop/src/main/java/me/aov/sellgui/dynashop/SellGUIDynaShop.java
package me.aov.sellgui.dynashop;

import me.aov.sellgui.SellGUIMain;
import org.bukkit.plugin.java.JavaPlugin;

public final class SellGUIDynaShop extends JavaPlugin {
    private DynaShopPriceProvider priceProvider;

    @Override
    public void onEnable() {
        SellGUIMain sellGUI = SellGUIMain.getInstance();
        if (sellGUI == null || sellGUI.getSellGUIAPI() == null) {
            getLogger().severe("SellGUI API is not available. Disabling SellGUI-DynaShop.");
            getServer().getPluginManager().disablePlugin(this);
            return;
        }

        priceProvider = new DynaShopPriceProvider(this);
        sellGUI.getSellGUIAPI().registerPriceProvider(priceProvider);
        getLogger().info("Registered ShopGUIPlus-DynaShop dynamic price provider with SellGUI.");
    }

    @Override
    public void onDisable() {
        SellGUIMain sellGUI = SellGUIMain.getInstance();
        if (sellGUI != null && sellGUI.getSellGUIAPI() != null && priceProvider != null) {
            sellGUI.getSellGUIAPI().unregisterPriceProvider(priceProvider);
        }
    }
}

File: SellGUI-DynaShop/src/main/kotlin/me/aov/sellgui/dynashop/SellGUIDynaShop.kt
package me.aov.sellgui.dynashop

import me.aov.sellgui.SellGUIMain
import org.bukkit.plugin.java.JavaPlugin

class SellGUIDynaShop : JavaPlugin() {
    private var priceProvider: DynaShopPriceProvider? = null

    override fun onEnable() {
        val sellGUI = SellGUIMain.getInstance()
        if (sellGUI == null || sellGUI.getSellGUIAPI() == null) {
            logger.severe("SellGUI API is not available. Disabling SellGUI-DynaShop.")
            server.pluginManager.disablePlugin(this)
            return
        }
        priceProvider = DynaShopPriceProvider(this)
        priceProvider?.let { sellGUI.getSellGUIAPI().registerPriceProvider(it) }
        logger.info("Registered ShopGUIPlus-DynaShop dynamic price provider with SellGUI.")
    }

    override fun onDisable() {
        val sellGUI = SellGUIMain.getInstance()
        val provider = priceProvider
        if (sellGUI?.getSellGUIAPI() != null && provider != null) sellGUI.getSellGUIAPI().unregisterPriceProvider(provider)
    }
}
